kolibri/content/urls.py

```diff
@@ -43,9 +44,10 @@ class Meta:
         fields = ['parent']
 
     def title_description_filter(self, queryset, value):
+        # only return the first 30 results to avoid major slow down
         return queryset.filter(
             Q(title__icontains=value) | Q(description__icontains=value)
-        )
+        )[0:30]
```

Comment:
I would suggest we use pagination to achieve this: http://www.django-rest-framework.org/api-guide/pagination/